app/controllers/event_notes_controller.rb

```diff
@@ -30,32 +30,32 @@ def update
     event_note = EventNote.find(event_note_id)
 
     # Save the state of the existing event note.
-    previous_event_note_revision = EventNoteRevision.where(
-      event_note_id: event_note_id
-    ).order(
-      :version
-    ).last
-
-    if previous_event_note_revision
-      version = previous_event_note_revision.version + 1
+    event_note_revision = build_revision(event_note)
+
+    unless event_note_revision.save
+      render json: { errors: event_note_revision.errors.full_messages }, status: 422
+    end
+
+    if event_note.update!(event_note_params)
+      render json: serialize_event_note(event_note)
     else
-      version = 1
+      render json: { errors: event_note.errors.full_messages }, status: 422
```

Comment:
We'll never reach this else clause because you're using `update!` which raises an exception if the update fails. Instead, you could use `update`.